### DISCUSSION TITLE
Firebase/Firestore connection - tutorial missing

### DISCUSSION THREAD

**Discussion by webwurm** [Budibase/budibase#6264] [Help] [answered]
I'd really love to use Budibase as a backend to my FlutterFlow-projects with their data in Firebase/Firestore, because it looks promissing!
Sadly I can't figure out how to connect to my firestore - and I can't find a tutorial anywhere, not even in the official documentation.

I tried it like this:
![Budibase - Firebase connection](https://user-images.githubusercontent.com/1053176/173029332-3fcd7bd4-fb02-4234-8f85-2c4f80548804.jpg)

But when I want to run a query (like in the pics; no idea, if it is correct for the collection "neuigkeiten", getting the field "nachricht"), I get the error:
"Query Error: Unexpected token i in JSON at position 1"

![Budibase - Firebase connection 3](https://user-images.githubusercontent.com/1053176/173029710-3ea86fd4-65bc-4e51-bf00-39ec93cd39bd.jpg)
![Budibase - Firebase connection 2](https://user-images.githubusercontent.com/1053176/173029714-c9902e5a-ab5b-4ff3-9559-810800b8d251.jpg)
 
I'd be happy if somebody could offer an explanation, how it works! Thanks!

**Comment by melohagan** [comment, marked as answer]
Hi @webwurm 

Docs added here: https://docs.budibase.com/docs/firestore

Also please note, there was a bug that has recently been fixed relating to the parsing of the private key: https://github.com/Budibase/budibase/pull/6269

This will be released within the next couple of weeks with the next hotfix build.

**Reply by webwurm** [reply]
Thank you very much!
After leaving "ServiceAccount" empty, I get the error:
Query Error: 16 UNAUTHENTICATED: Failed to retrieve auth metadata with error: error:0909006C:PEM routines:get_name:no start line

Can that be connected with the bug you were talking about?
Thanks!

**Reply by melohagan** [reply]
Yes that is the bug. It isn't correctly parsing the private key, but that will be fixed when next released!

**Reply by webwurm** [reply]
Thanks, it works perfectly now!
